LinqExercises2/Excersie11.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections.Generic;
+
+namespace Generators2
+{
+    public static class GeneratorsExercise2
+    {
+        // Write a method that implements a bounded 
+        // generator, providing the first n values
+        // for the sequence of integers that follows
+        // the pattern prev + 2, prev x 2, prev + 2, etc.
+        // where n is passed in to the method.
+        //
+        // The method should be called AddTwoThenDouble()
+        // and should take a single integer parameter.
+
+        // public static ... AddTwoThenDouble(int n)
+        // {
+        // }
+        public static IEnumerable<int> AddTwoThenDouble(int n)
+        {
+            int evenOrOdd = 0;
+            int nextVal = 0;
+
+            for (int i = 0; i < n; i++)
+            {
+                nextVal = evenOrOdd % 2 == 0 ? nextVal + 2 : nextVal * 2;

```

Comment:
1. you can decide whether it's even or odd by checking the index i
2. ternary operators with long expressions makes it harder to understand. I prefer using it only when the expressions are simple, like just a variable or a  method call, for example:
```CSharp
y = isValid(x) ? Deserialize(x) : defaultValue;
```
Otherwise I would use a simple if statement
```CSharp
if (i % 2 == 0)
{
    nextVal += 2;
}
else
{
    nextVal *= 2;
}
```

Reply:
For the first comment, yes, the index of the for loop is sufficient to check for odd or even numbers.
But for the second comment, I thought that if it was one line, we would use this format
I will modify the solution

Reply:
@Mo7ammadAbuSafat there are no general rules, we're just writing what we think would be easier to read, and that's why people are doing code reviews, to get some insights about the code from different perspective.

Reply:
@MazenAmria all right, I got it